python/ray/data/_internal/planner/plan_udf_map_op.py

```diff
@@ -127,6 +128,69 @@ def fn(block: Block) -> Block:
     )
 
 
+def plan_streaming_repartition_op(
+    op: Project,
+    physical_children: List[PhysicalOperator],
+    data_context: DataContext,
+) -> MapOperator:
+    assert len(physical_children) == 1
+    input_physical_dag = physical_children[0]
+
+    max_num_rows_per_block = op.max_num_rows_per_block
+
+    def fn(block: Block) -> List[Block]:
+        accessor = BlockAccessor.for_block(block)
+        num_rows_in_block = accessor.num_rows()
+
+        # If the number of rows in block is within the allowed max_num_rows_per_block, nothing to do.
+        if num_rows_in_block <= max_num_rows_per_block:
+            return [block]
+
+        block_list = []
+        cur_block_builder = DelegatingBlockBuilder()
+        start_idx = 0
+
+        # Iterate through the block, partitioning it into smaller blocks.
+        while start_idx < num_rows_in_block:
+            remaining_rows = num_rows_in_block - start_idx
+            cur_rows = cur_block_builder.num_rows()
+
+            # Determine the slice range for the next partition.
+            end_idx = start_idx + min(remaining_rows, max_num_rows_per_block - cur_rows)
+            cur_block_builder.add_block(accessor.slice(start_idx, end_idx, copy=True))
+
+            # If the current block reaches the size limit, finalize and store it.
+            if cur_block_builder.num_rows() == max_num_rows_per_block:
```

Comment:
this condition isn't needed. Because the current block will either have `max_num_rows_per_block` or be the last block. 
Also, the builder isn't needed either. Because we are not combining slices across multiple input blocks. We can just return the sliced block.